src/scipp/coords/transform_coords.py

```diff
@@ -75,17 +82,30 @@ def transform_coords(x: Union[DataArray, Dataset],
                       keep_intermediate=keep_intermediate,
                       keep_inputs=keep_inputs,
                       quiet=quiet)
-    targets = {targets} if isinstance(targets, str) else set(targets)
-    if isinstance(x, DataArray):
-        return _transform_data_array(x,
-                                     targets=targets,
-                                     graph=Graph(graph),
-                                     options=options)
+    if targets is None:
+        targets = []
+    if graph is None:
+        graph = {}
```

Comment:
The whole argument handling seems very complicated and has a lot of interconnected rules. While simple use cases are straight forward, I am worried that more complex cases can get messy.

Reply:
The most complex example I can think of is giving `targets`, `graph`, *and* `kwargs`:

```python
da.transform_coords('x', y=func, graph=base_graph)
```
which would compute `x` and `y`, based on a graph made from `base_graph` and `{'y':func}`. Is this the messy complex cases you have in mind, or is there another example?